src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -988,14 +1065,22 @@ static void Main()
             var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
 
             var def = nodes.OfType<DefaultLiteralSyntax>().Single();
-            Assert.Null(model.GetTypeInfo(def).Type);
+            Assert.Equal("System.Int16", model.GetTypeInfo(def).Type.ToTestDisplayString());
             Assert.Null(model.GetSymbolInfo(def).Symbol);
             Assert.Null(model.GetDeclaredSymbol(def));
+            Assert.Equal("System.Int16", model.GetTypeInfo(def).ConvertedType.ToTestDisplayString());
+            Assert.Null(model.GetSymbolInfo(def).Symbol);
+            Assert.Equal((short)0, model.GetConstantValue(def).Value);
+            Assert.True(model.GetConversion(def).IsIdentity);
```

Comment:
Yes, in terms of conversions, the behavior is similar with `null` (there is an identity conversion on the literal).
The main difference with `null` as we already discussed is that we give `default` a type.

```C#
        [Fact]
        public void ExplicitCastOnNull()
        {
            string source = @"
class Base { }
class Derived : Base 
{
    static void Main()
    {
        Base x = (Derived)null;
        System.Console.Write(x);
    }
}
";

            var comp = CreateCompilationWithMscorlib(source, parseOptions: TestOptions.ExperimentalParseOptions, options: TestOptions.DebugExe);
            comp.VerifyDiagnostics();

            var tree = comp.SyntaxTrees.First();
            var model = comp.GetSemanticModel(tree);
            var nodes = tree.GetCompilationUnitRoot().DescendantNodes();

            var def = nodes.OfType<LiteralExpressionSyntax>().Single();
            Assert.Equal(null, model.GetTypeInfo(def).Type);
            Assert.Null(model.GetSymbolInfo(def).Symbol);
            Assert.Null(model.GetDeclaredSymbol(def));
            Assert.Equal(null, model.GetTypeInfo(def).ConvertedType);
            Assert.Null(model.GetSymbolInfo(def).Symbol);
            Assert.True(model.GetConstantValue(def).HasValue);
            Assert.Equal(null, model.GetConstantValue(def).Value);
            Assert.True(model.GetConversion(def).IsIdentity);

            var conversionSyntax = nodes.OfType<CastExpressionSyntax>().Single();
            var conversionTypeInfo = model.GetTypeInfo(conversionSyntax);
            Assert.Equal("Derived", conversionTypeInfo.Type.ToTestDisplayString());
            Assert.Equal("Base", conversionTypeInfo.ConvertedType.ToTestDisplayString());
            Assert.True(model.GetConstantValue(conversionSyntax).HasValue);
            Assert.Equal(null, model.GetConstantValue(conversionSyntax).Value);
            Conversion conversion = model.GetConversion(conversionSyntax);
            Assert.False(conversion.IsNumeric);
            Assert.True(conversion.IsImplicit);
        }
```